services/search/src/search/routes/filter.py

```diff
@@ -0,0 +1,278 @@
+                    supported_columns, unsupported_columns = get_supported_unsupported_columns(
+                        features,
+                    )
+                with StepProfiler(method="filter_endpoint", step="execute filter query"):
+                    num_rows_total, pa_table = execute_filter_query(
+                        index_file_location=index_file_location,
+                        columns=supported_columns,
+                        where=where,
+                        limit=length,
+                        offset=offset,
+                    )
+                with StepProfiler(method="search_endpoint", step="clean cache randomly"):
+                    clean_cached_assets_randomly(
+                        clean_cache_proba=clean_cache_proba,
+                        dataset=dataset,
+                        cached_assets_directory=cached_assets_directory,
+                        keep_first_rows_number=keep_first_rows_number,
+                        keep_most_recent_rows_number=keep_most_recent_rows_number,
+                        max_cleaned_rows_number=max_cleaned_rows_number,
+                    )
+                with StepProfiler(method="filter_endpoint", step="create response"):
+                    response = create_response(
+                        dataset=dataset,
+                        config=config,
+                        split=split,
+                        cached_assets_base_url=cached_assets_base_url,
+                        cached_assets_directory=cached_assets_directory,
+                        s3_client=s3_client,
+                        cached_assets_s3_bucket=cached_assets_s3_bucket,
+                        cached_assets_s3_folder_name=cached_assets_s3_folder_name,
+                        pa_table=pa_table,
+                        offset=offset,
+                        features=features,
+                        unsupported_columns=unsupported_columns,
+                        num_rows_total=num_rows_total,
+                    )
+                with StepProfiler(method="filter_endpoint", step="generate the OK response"):
+                    return get_json_ok_response(content=response, max_age=max_age_long, revision=revision)
+            except Exception as e:
+                error = e if isinstance(e, ApiError) else UnexpectedError("Unexpected error.", e)
+                with StepProfiler(method="filter_endpoint", step="generate API error response"):
+                    return get_json_api_error_response(error=error, max_age=max_age_short, revision=revision)
+
+    return filter_endpoint
+
+
+def execute_filter_query(
+    index_file_location: str, columns: list[str], where: str, limit: int, offset: int
+) -> tuple[int, pa.Table]:
+    con = duckdb.connect(database=index_file_location, read_only=True)
+    # TODO: Address possible SQL injection CWE-89
+    filter_query = FILTER_QUERY.format(columns=",".join(columns), where=where, limit=limit, offset=offset)
```

Comment:
OK. We really don't look at what is passed in the where clause. I understand that it would be complex, but... we don't check it's against a supported column type, or if the query makes sense, or if it's dangerous...

I'm no specialist, so I would prefer other opinions on this. cc @AndreaFrancis maybe

Reply:
I think we are good for now but maybe wrap this part in a try/except block to provide better feedback to the client.
Also, consider that we load the db file in read-only mode.
Regarding dangerous code, I'm not sure if we can use the where wildcard (?) with prepared statements https://duckdb.org/docs/api/c/prepared.html (avoid SQL Injection).

Reply:
I implemented the `try... except` around the SQL query, so that we return an `InvalidParameterError`.

Reply:
In relation with the validation of `where` parameter, see `validate_where_parameter` done when getting `where` from `response` in the body of `create_filter_endpoint`.